server/controllers/userController.js

```diff
@@ -65,3 +66,38 @@ userController.verifyUser = (req, res, next) => {
       // *** BUT WE NEED LOGIC IN LOGIN PAGE TO NOT ALLOW BLANK PASSWORD SUBMISSION
       // USER CAN BE VERIFIED
 };
+
+userController.jwt = (req, res, next) => {
+  try {
+    const userObject = jwt_decode(req.body.credential);
+    res.locals.googleEmail = userObject.email;
+    return next();
+  } catch(error) {
+    return next({
+      log: 'Error occurred in userController.jwt middleware',
+      status: 500,
+      message: { error },
+    });
+  }
+}
+
+userController.verifyOauth = async (req, res, next) => {
+  try {
+    const queryExistingUser = `SELECT * FROM users WHERE email = ${res.locals.googleEmail}`;
+    const queryResult = await db.query(queryExistingUser);
+    console.log('query results: ', queryResult);
+
+    if(queryResult.rows.length === 0) {
+      const queryStr = `INSERT INTO users (id, username, email, password) VALUES (DEFAULT, $1, $2, $3) RETURNING id`;
+      const newGoogleUser = ['google user', email, 'google password'];
+      const userId = await db.query(queryStr, newGoogleUser);
```

Comment:
Trying to query to verify if a user exist before inserting can cause race conditions. It is best to do an "upsert", meaning only insert if the information does not exist in the database.

Look into `ON CONFLICT` key word and perhaps ask Eileen who has some experience with that as well